ha/core/event_manager/event_manager.py

```diff
@@ -57,10 +84,104 @@ def _validate_component(self, component: str) -> None:
         if component.lower() not in SUBSCRIPTION_LIST:
             raise InvalidComponent(f"Invalid component {component}, not part of subscription list.")
 
-    def _validate_events(self, events: list) -> None:
-        """
-        Raise error InvalidEvent if event is not valid.
+    def subscribe(self, component : str, events : list):
+        '''
+           Register all the events for the notification. It maintains
+           list of events registered by the components. It maps the event
+           name with component name and store it in the consul. \
+           Also, maps the component with event
 
+           Args:
+            component (str): Component Name
+            events (list): Events.
+        '''
+        Log.info(f'Received a subscribe request from {component} for event: {events}')
+        # self._validate_component(component)
+        self._validate_events(events)
+        if component: _create_message_type(component)
+        resource_type = events.resource_type
+        states = events.states
+        self._store_component_key(f'{const.COMPONENT_KEY}/{component}', resource_type, states)
+        self._store_event_key(f'{const.EVENT_KEY}', resource_type, states, comp=component)
+        Log.info('Subscription request is successfully stored into consul')
+
+    def _store_component_key(self, key: str, resource_type: str, val: list) -> None:
```

Comment:
_store_component_key
_store_event_key
Both look same can we refactor to make one fun only

Reply:
I think its good to separate the logic, this looks similar. But keys are different.
One key is of type:
cortx/ha/v1/events/enclosure:interface:sas/failed:["motr"]

One key is of type:
cortx/ha/v1/events/subscribe/hare:["node:os:memory_usage/failed", "node:interface:nw/online"]

Which require some different level of logic. May be if its a need, a TODO can be applied here.

Reply:
There is some code duplication but will handle in refactor